scripts/mcp_server.py

```
@@ -344,31 +522,257 @@ def h_panel_ingest(args):
+                # (.bak, like .prev, is not *.json, so it never enters the attestation.)
+                # (Codex, 52c686f; CodeRabbit, 60cb2c3.)
+                try:
+                    before_mtime = vf.stat().st_mtime_ns
+                    stash_bytes = vf.read_bytes()
+                    cand_bak = vf.parent / (vf.name + ".bak")
+                    cand_bak.write_bytes(stash_bytes)
+                    stash_backup = cand_bak
+                except OSError:
+                    raise ToolError("cannot snapshot the prior verdict.json to guarantee a restore "
+                                    "(it could be neither moved aside nor durably backed up) — refusing "
+                                    "to aggregate so a prior verdict is never lost")
```

Ruff (0.16.3)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)